### PR TITLE
Send demand control costs to Studio via protobuf limits field

### PR DESCRIPTION
**Pull request by tninesling**
If you are an Apollo GraphOS user, field usage and lengths can be sent to apollo studio without requiring subgraphs to support FTV1.
This data is not currently displayable in GraphOS Studio.

This is an experimental feature and must be enabled via:

```yaml
telemetry:
  apollo:
    experimental_local_field_metrics: true
```

The reason for this is there is currently a slight performance impact from using this.

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [X] Unit Tests
    - [x] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Review by BrynCooke — APPROVED** [review]

**Review by BrynCooke — CHANGES_REQUESTED** [review]
Code looks great, but there looks to be perf issues.

**Review by BrynCooke — APPROVED** [review]

File: apollo-router/src/plugins/telemetry/metrics/apollo/list_length_histogram.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
+use std::ops::AddAssign;
+
+use hdrhistogram::Histogram;
+use serde::ser::SerializeSeq;
+use serde::Serialize;
+
+use super::studio::MAX_HISTOGRAM_BUCKETS;
+
+/// A histogram for recording lengths of list fields in GraphQL responses. This implementation is clamped to a maximum
+/// of 386 buckets, a restriction imposed by Studio. The buckets roughly follow order of magnitude of the recorded value,
```
Comment (garypen) [inline apollo-router/src/plugins/telemetry/metrics/apollo/list_length_histogram.rs:10]: I note that duration histogram uses 383 and that was taken from the TS implementation. Which one is correct?

**Review by garypen — COMMENTED** [review]
Do we need the new histogram types or could we re-purpose DurationHistogram somehow into a general purpose thing?

File: apollo-router/src/plugins/telemetry/metrics/apollo/list_length_histogram.rs (hunk 1 above)
Comment (BrynCooke) [reply apollo-router/src/plugins/telemetry/metrics/apollo/list_length_histogram.rs:10]: I've fixed this in the histogram refactor code.

**Pull request merged**